packages/graphql/src/translate/cypher-builder/references/Relationship.ts

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { CypherContext } from "../CypherContext";
+import { escapeLabel, padLeft } from "../utils";
+import { Node } from "./Node";
+import { Param } from "./Param";
+import { CypherReference } from "./Reference";
+import { serializeParameters } from "./utils";
+
+export type RelationshipInput = {
+    source: Node;
+    target: Node;
+    type?: string;
+    parameters?: Record<string, Param<any>>;
+    directed?: boolean;
+};
+
+export class Relationship implements CypherReference {
```

Comment:
I will admit I'm a bit confused by `CypherReference` - in one place this defined the behaviour of a parameter, and here a relationship? They seem like two completely different concepts. Surely a relationship has more to do with a node than it does a parameter?